app/src/main/java/cellscanner/wowtor/github/com/cellscanner/LocationService.java

```diff
@@ -109,7 +108,7 @@ public void onCreate() {
         Runnable timer = new Runnable() {
             @Override
             public void run() {
-                if (running) {
+                if (inRecordingState(getApplicationContext())) {
```

Comment:
why this check? shouldn't the timer be stopped in the first place when not in recording state?

Reply:
Correct, but the timer check is not in this pull request. Will remove

Reply:
Check is removed. The will result in unwanted behaviour when not merged with "feature/stop_thread_after_stop_recording"